portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx

```diff
@@ -0,0 +1,341 @@
+// This file is part of MinIO Console Server
+// Copyright (c) 2021 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { useEffect, useState, Fragment } from "react";
+import { connect } from "react-redux";
+import { createStyles, Theme, withStyles } from "@material-ui/core/styles";
+import { Button } from "@material-ui/core";
+import Grid from "@material-ui/core/Grid";
+import TextField from "@material-ui/core/TextField";
+import InputAdornment from "@material-ui/core/InputAdornment";
+import SearchIcon from "@material-ui/icons/Search";
+import FileCopyIcon from "@material-ui/icons/FileCopy";
+import Moment from "react-moment";
+import {
+  RemoteBucket,
+  RemoteBucketList,
+  HasPermissionResponse,
+} from "../types";
+import { CreateIcon } from "../../../../icons";
+import { AppState } from "../../../../store";
+import { addBucketOpen, addBucketReset } from "../actions";
+import { setErrorSnackMessage } from "../../../../actions";
+import {
+  actionsTray,
+  containerForHeader,
+  searchField,
+} from "../../Common/FormComponents/common/styleLibrary";
+import api from "../../../../common/api";
+import TableWrapper from "../../Common/TableWrapper/TableWrapper";
+import AddBucket from "./AddBucket";
+import DeleteBucket from "./DeleteBucket";
+import PageHeader from "../../Common/PageHeader/PageHeader";
+import BulkReplicationModal from "./BulkReplicationModal";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    seeMore: {
+      marginTop: theme.spacing(3),
+    },
+    paper: {
+      display: "flex",
+      overflow: "auto",
+      flexDirection: "column",
+    },
+
+    addSideBar: {
+      width: "320px",
+      padding: "20px",
+    },
+    tableToolbar: {
+      paddingLeft: theme.spacing(2),
+      paddingRight: theme.spacing(0),
+    },
+    minTableHeader: {
+      color: "#393939",
+      "& tr": {
+        "& th": {
+          fontWeight: "bold",
+        },
+      },
+    },
+    ...actionsTray,
+    ...searchField,
+    ...containerForHeader(theme.spacing(4)),
+  });
+
+interface IListBucketsProps {
```

Comment:
Would be better to rename this as `IListRemoteBucketsProps`

```suggestion
interface IListRemoteBucketsProps {
```